tests/phpunit/tests/blocks/editor.php

```diff
@@ -720,4 +720,21 @@ public function data_block_editor_rest_api_preload_adds_missing_leading_slash()
 			),
 		);
 	}
+
+	/**
+	 * @ticket 61641
+	 */
+	public function test_get_block_editor_settings_block_bindings_sources() {
+		$block_editor_context              = new WP_Block_Editor_Context();
+		$settings                          = get_block_editor_settings( array(), $block_editor_context );
+		$registered_block_bindings_sources = get_all_registered_block_bindings_sources();
+
+		foreach ( $registered_block_bindings_sources as $name => $properties ) {
+			$registered_properties = get_object_vars( $properties );
+			// Remove name property as it is not part of the settings.
+			unset( $registered_properties['name'] );
```

Comment:
Ideally, we should probably remove the `name` property from the object as it is already the `key`. If I am not mistaken, that could be done changing this property to `private`: [link](https://github.com/SantosGuillamot/wordpress-develop/blob/745509ab27e82497c757b8341b671cc326084ef3/src/wp-includes/class-wp-block-bindings-source.php#L29). However, I assume it could cause backwards compatibility problems if people are relying on it.

Any thoughts?

Reply:
Can you hardcode the list of expected values instead? That would be easier to follow. If there are ever any changes, it will be applied manually and go through the review process. With the current implementation, there is a risk some additional properties will be added but not necessarily need to be exposed and `name` is a good example here.

Reply:
Now that I think more about it, you're right and this probably doesn't make much sense. The registered sources have other properties like `get_value_callback` that aren't exposed in the editor settings. I will hardcode the list of expected values.

Reply:
I made the suggested changes in [this commit](https://github.com/WordPress/wordpress-develop/pull/7020/commits/ac767be71bd9be25dd95a767767eb9721eb3750b).

Reply:
I really think there should be no loops and conditional checks in the test, but something like:

https://github.com/WordPress/wordpress-develop/blob/f003fd07ba5da8dd0266715d3874250040eab605/tests/phpunit/tests/rest-api/rest-block-type-controller.php#L531-L552

or

https://github.com/WordPress/wordpress-develop/blob/f003fd07ba5da8dd0266715d3874250040eab605/tests/phpunit/tests/blocks/editor.php#L206-L249

Reply:
Thanks for the links 🙂 I've tried a different approach in [this commit](https://github.com/WordPress/wordpress-develop/pull/7020/commits/2b2bed58356bdfa66e3e7e8682c28d1caa2fcb1e). Basically, I'm registering two new sources in the test and check they are exposed in the settings. I created two of them to test that `usesContext` can be empty, but I can remove it if it is too noisy.

Apart from that, it is checking right now the count manually as the rest of the tests: [link](https://github.com/WordPress/wordpress-develop/pull/7020/files#diff-f757314c3e344098b5cd8a11b80feda25959ed87435042bd1f47e108c035ea7aR747). Do you think it would be better to get the value from `getAllBlockBindingsSources`?

Please, let any more feedback you have. Happy to iterate on it.

Reply:
Two new sources make perfect sense. I didn't think about it, but this is even better as these added for the test are fully predictable 👍🏻